internal/rootcoord/dml_channels.go

```diff
@@ -82,13 +82,12 @@ func (d *dmlChannels) GetNumChannels() int {
 func (d *dmlChannels) Broadcast(chanNames []string, pack *msgstream.MsgPack) error {
 	for _, chanName := range chanNames {
 		// only in-use chanName exist in refcnt
-		if _, ok := d.refcnt.Load(chanName); ok {
-			v, _ := d.pool.Load(chanName)
-			if err := (*(v.(*msgstream.MsgStream))).Broadcast(pack); err != nil {
-				return err
+		if _, ok1 := d.refcnt.Load(chanName); ok1 {
+			if v, ok2 := d.pool.Load(chanName); ok2 {
+				if err := (*(v.(*msgstream.MsgStream))).Broadcast(pack); err != nil {
+					return err
+				}
 			}
-		} else {
-			return fmt.Errorf("channel %s not exist", chanName)
 		}
```

Comment:
If the channel name is not valid, an error should be returned. I suggest that an error returns if either map fails to load with provided channel name.